File: Python/quicksort.py
import random

def quick_sort(arr):
	if (len(arr) <= 1):
		return arr
	else:
		x = arr[0]
		xs = arr[1:len(arr)]
		left = list(filter(lambda y: y < x, xs))
		right = list(filter(lambda y: y >= x, xs))
		return quick_sort(left) + [x] + quick_sort(right)

arr = [random.randrange(100) for x in range(0, 100)]
print(quick_sort(arr))

File: Ruby/quicksort.rb
def quicksort(arr)
	if arr.size <= 0
		return arr
	end
	x = arr[0]
	right = Array.new
	left = Array.new
	for i in 1..arr.size-1
		if arr[i] <= x
			left.push(arr[i])
		else
			right.push(arr[i])
		end
	end

	left = quicksort(left)
	right = quicksort(right)
	return left + [x] + right
end

arr = []
100.times{ arr << rand(100) }
p(quicksort(arr))
